### PR TITLE
Remove bus from dashboard api

### PR DESCRIPTION
**Pull request by idafurjes**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/api/dashboard_test.go [hunk 1]
```diff
@@ -43,13 +43,14 @@ func TestGetHomeDashboard(t *testing.T) {
...
-			HomeDashboardId: 0,
-		}
-		return nil
-	})
+	// hs.Bus.AddHandler(func(_ context.Context, query *models.GetPreferencesWithDefaultsQuery) error {
+	// 	query.Result = &models.Preferences{
+	// 		HomeDashboardId: 0,
+	// 	}
+	// 	return nil
+	// })
```
Comment (zserge) [inline pkg/api/dashboard_test.go:53]: 😉

Comment (idafurjes) [reply pkg/api/dashboard_test.go:53]: this test is not failing

File: pkg/services/sqlstore/dashboard.go [hunk 2]
```diff
@@ -375,7 +375,7 @@ func makeQueryResult(query *search.FindPersistedDashboardsQuery, res []Dashboard
 	}
 }
 
-func GetDashboardTags(ctx context.Context, query *models.GetDashboardTagsQuery) error {
+func (ss *SQLStore) GetDashboardTags(ctx context.Context, query *models.GetDashboardTagsQuery) error {
```
Comment (zserge) [inline pkg/services/sqlstore/dashboard.go:378]: I think this can be removed as well?

File: pkg/services/sqlstore/dashboard.go [hunk 3]
```diff
@@ -601,7 +601,7 @@ func GetDashboardSlugById(ctx context.Context, query *models.GetDashboardSlugByI
 	return nil
 }
 
-func GetDashboardsBySlug(ctx context.Context, query *models.GetDashboardsBySlugQuery) error {
+func (ss *SQLStore) GetDashboardsBySlug(ctx context.Context, query *models.GetDashboardsBySlugQuery) error {
```
Comment (zserge) [inline pkg/services/sqlstore/dashboard.go:604]: I think this can be removed as well?

File: pkg/services/sqlstore/mockstore/mockstore.go [hunk 4]
```diff
@@ -416,9 +425,19 @@ func (m *SQLStoreMock) SaveDashboard(cmd models.SaveDashboardCommand) (*models.D
 }
 
 func (m *SQLStoreMock) GetDashboard(ctx context.Context, query *models.GetDashboardQuery) error {
+	query.Result = m.ExpectedDashboard
+	return m.ExpectedError
+}
+
+func (m *SQLStoreMock) GetDashboardsBySlug(ctx context.Context, query *models.GetDashboardsBySlugQuery) error {
```
Comment (zserge) [inline pkg/services/sqlstore/mockstore/mockstore.go:432]: Probably can be removed?

File: pkg/services/sqlstore/store.go [hunk 5]
```diff
@@ -105,6 +105,8 @@ type Store interface {
 	RemoveOrgUser(ctx context.Context, cmd *models.RemoveOrgUserCommand) error
 	SaveDashboard(cmd models.SaveDashboardCommand) (*models.Dashboard, error)
 	GetDashboard(ctx context.Context, query *models.GetDashboardQuery) error
+	GetDashboardsBySlug(ctx context.Context, query *models.GetDashboardsBySlugQuery) error
```
Comment (zserge) [inline pkg/services/sqlstore/store.go:108]: Also, can be removed?

File: pkg/api/dashboard_permission_test.go [hunk 6]
```diff
@@ -23,21 +23,20 @@ import (
...
-				bus.AddHandler("test", func(ctx context.Context, query *models.GetDashboardQuery) error {
-					return models.ErrDashboardNotFound
-				})
-			}
-			mock := mockstore.NewSQLStoreMock()
+			mockSQLStore.ExpectedError = models.ErrDashboardNotFound
 			loggedInUserScenarioWithRole(t, "When calling GET on", "GET", "/api/dashboards/id/1/permissions",
 				"/api/dashboards/id/:dashboardId/permissions", models.ROLE_EDITOR, func(sc *scenarioContext) {
-					setUp()
+					// setUp()
```
Comment (zserge) [inline pkg/api/dashboard_permission_test.go:36]: Needed or not?

**Review by zserge — APPROVED** [review]

**Pull request merged**